bundles/org.eclipse.swt/Eclipse SWT Browser/win32/org/eclipse/swt/browser/Edge.java

```diff
@@ -557,13 +588,19 @@ public boolean execute(String script) {
 	// Feature in WebView2. ExecuteScript works regardless of IsScriptEnabled setting.
 	// Disallow programmatic execution manually.
 	if (!jsEnabled) return false;
-
+	waitForWebviewInitialization();
 	IUnknown completion = newCallback((long result, long json) -> COM.S_OK);
-	int hr = webView.ExecuteScript(stringToWstr(script), completion);
+	int hr = lastWebViewTask.join().ExecuteScript(stringToWstr(script), completion);
 	completion.Release();
 	return hr == COM.S_OK;
 }
 
+private void waitForWebviewInitialization() {
+	while(!lastWebViewTask.isDone() || !lastWebView_2Task.isDone()) {
```

Comment:
Don't you also have to wait for `webView_11` ?

Reply:
its not relevant for the deadlock :) its only used for the context menu callback .Do you think its worth adding here?

Reply:
If it doesn't bring any value then there's no need to add it. I have to admit that I cannot make that call so I'll leave it to you. Let me know what you decide.

Reply:
All those `ICoreWebView2_*` references point to the exactly same instance, they are just exposing more or less methods of the same.